Add initial version of bitcoin primer slides

dariothuerkauf opened the pull request:
# Description

* Addition of first version of tex file for lecture "Bitcoin Primer" including the respective images used and figures created.
* Figures created: bitcoin-system.tex, double-spend.tex, transaction-capacity.tex, transaction-manipulation.tex
* Images added: block_1, blocks_3, consensus_problem
* Config changes: added tikzlibraries (decorations.pathreplacing, decorations.markings, positioning), added captionsetup


## Type of Change

- [x ] Major feature (New files or assets, large changes to files)

# Checklist:

- [x] I have performed a self-review of my own changes
- [x] I have made sure that any changed LaTeX files compile properly
- [x] I have assigned at least one reviewer

Rothj commented on slides/bitcoin-primer.tex:
```diff
@@ -0,0 +1,137 @@
+Transactions are bundled into blocks \\
+\includegraphics[width=6cm]{../assets/images/block_1.png} \\
+\uncover<2->{
+Those blocks are sequentially linked $\rightarrow $ blockchain \\
+\includegraphics[width=6cm]{../assets/images/blocks_3.png}
+} 
```
I think it would make sense to insert a slide after this to explain how the blocks are linked in more detail (e.g. with the figure of three blocks showing how each use the ID of the preceding block).

In addition to that, I think the original slides on the "bitcoin miner" were quite nice and would fit well at this stage as they contain nice visualizations and lead up to the next slide.

Chainomics replied:
@dariothuerkauf I tend to agree with @Rothj. While this slide deck serves as a high-level overview, it may still be useful to add some more context and get a little closer to the original slides.

dariothuerkauf replied:
I agree. I will add the original slides on the "bitcoin miner" in the next commit.

Rothj reviewed (commented):
Well done! I have one general points on the part on transaction consensus: 
This part is much shorter than in the previously used slides. This is nice I think! But it leads to much more details only being explained in the specific part on trx-consensus. We should keep that in mind when doing these slide sets. In addition to that there might be some parts of the course which require the parts left out as a prerequisite (e.g. Cryptography: why are we talking about hash values?). I guess there is a trade of between a compact overview and already covering enough topics so later chapters can be understood more easily. However, I think depending on what will be said along the slides, the slides already balance this trade-off quite well!

Chainomics commented:
@dariothuerkauf Please check my commits. I have added an additional figure, separated the figure environment from the figure code, and added an additional slide. Moreover, I changed some of the text. @Rothj please review as well. Cheers!

dariothuerkauf commented on slides/bitcoin-primer.tex:
```diff
@@ -0,0 +1,137 @@
+%%%
+\begin{frame}{Transaction consensus}
+Who can broadcast a new block? \\ \vspace{1.5em}
+
```
Fair point, how about adding the line "**Goal:** Deciding which (legitimate) transactions are valid." ?

dariothuerkauf commented:
@Chainomics @Rothj Thank you for your feedback and help! I added the bitcoin miner and the simplified blockchain structure slides. In last year's lecture you also introduced proof-of-work, competing blocks, attacks on the blockchain and the coinbase transaction. Do you think any of these topics should also be included here?

Chainomics commented:
@dariothuerkauf Please add a short introduction to the world of cryptocurrencies by Aleks and myself to the recommended reading list/references at the end of the slide deck. Also: Please review my changes.

golmit commented on slides/bitcoin-primer.tex:
```diff
@@ -0,0 +1,216 @@
+%%%
+
+%%%
+\begin{frame}{Transaction Capacity}
+\textbf{Goal:} ensuring the ability to initiate a transaction  \\
+\vspace{1em}
+\uncover<2->{Bitcoin network 
+        \begin{itemize}
+		\item {consists of all users}
+		\item {primary communication path}
+		%\item<2->{completely decentral}
+		\end{itemize}
+		}
+\vspace{1em}
+\uncover<3->{Bitcoin protocol
+        \begin{itemize}
+        \item<3->{defines manner in which communication has to take place (ruleset)}
+        \end{itemize}
+        }     
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Transaction Capacity}  %example banking vs. decentral system?
+	
+	\begin{figure}[h!]
+		\center
+		\input{../assets/figures/transaction-capacity.tex}
+	\end{figure}\vspace{1em}
+
+\uncover<2->{       
+	Peer-to-Peer Network: 
+        \begin{itemize}
+		\item<1-> permissionless
+		\item<1-> censorship-resistant
+		\item<1-> No special privileges
+		\end{itemize}
+		}
+\end{frame}
```
I suggest that we remove the bullet points and reduce the amount of text for the first trx-capacity slide. Nonetheless, these things are important and should be said during the video. 

```suggestion
\begin{frame}{Transaction Capacity}
\textbf{Goal:} ensuring the ability to initiate a transaction  \\
	\begin{figure}[h!]
		\center
		\input{../assets/figures/transaction-capacity.tex}
	\end{figure}\vspace{1em}

\uncover<2->{       
	Peer-to-Peer Network: 
        \begin{itemize}
		\item<1-> Permissionless
		\item<1-> Censorship-resistant
		\item<1-> No special privileges
		\end{itemize}
		}
\end{frame}
```
Alternatively, use one more uncover command:
1. Goal line
2. Figure
3. P2P network summary

I really like the following slides about transaction legitimacy and consensus!

golmit commented on slides/bitcoin-primer.tex:
```diff
@@ -0,0 +1,216 @@
+
+
+%%%
+\begin{frame}{Bitcoin Miner}
+Network participants, who decide to allocate computing power in order to create new blocks are called Bitcoin Miners.
+\center
+\includegraphics[width=6cm]{../assets/images/miner.png}
+
+\begin{itemize}
+\item{Miner chooses the status quo on which to base its block (most recent version of Blockchain)}
+\item{Miner chooses a subset of the transactions from its queue (mempool) to create a block, as long as they...}
+	\begin{itemize}
+	\item{\dots are legitimate}
+	\item{\dots do not conflict with other transactions}
+	\item{\dots do not exceed block size limitation}
+	\end{itemize}
+\end{itemize}
+\end{frame}
```
I would reduce the text on this slide, but I'm not sure how this can easily be done. Maybe the figure and one statement are sufficient.

golmit commented on slides/bitcoin-primer.tex:
```diff
@@ -0,0 +1,216 @@
+\end{frame}
+
+%%%
+\begin{frame}{Definition}
+Some key aspects of Bitcoin: \vspace{1em}
+
+	\begin{enumerate}
+		\item<2-> Competitive creation
+		\item<3-> Virtual representation
+		\item<4-> Decentralized management
+	\end{enumerate}
+	\vspace{1em}	
+\uncover<5->{Bitcoin is a decentralized data structure.} \\	\vspace{1em}
+\uncover<6->{The system is maintained by its participants and works in the absence of centralized third parties.} 
+\end{frame}
```
There will be a video which covers the monetary control structures (of Bitcoin) right before the Bitcoin Primer. I suggest that we then either:
- remove the Definition slide and summarize its content during the introduction of the video, or
- that we replace the bullet points on this slide with the figure from the monetary control structures slide deck.

Anyway, let's leave it as is and fix it as soon as the monetary control structure slides are ready.

Chainomics reviewed (approved):
Thank you @dariothuerkauf! As proposed by @golmit we can edit the control structure slide, once the other slide deck is ready.
